eventsourcing/src/main/java/org/axonframework/eventsourcing/annotation/ConstructorBasedEventSourcedEntityCreator.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.annotation;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.common.infra.ComponentDescriptor;
+import org.axonframework.common.infra.DescribableComponent;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+import static org.axonframework.common.ConstructorUtils.getConstructorFunctionWithOptionalArgumentClass;
+
+/**
+ * {@link EventSourcedEntityCreator} implementation which uses a constructor to create a new instance of an entity. The
+ * constructor can either have a single argument of the same type as the identifier, or no arguments at all.
+ *
+ * @param <M> The type of the entity to create.
+ * @author Mitchell Herrijgers
+ * @since 5.0.0
+ */
+public class ConstructorBasedEventSourcedEntityCreator<M>
+        implements EventSourcedEntityCreator<Object, M>, DescribableComponent {
+
+    private final Class<M> entityType;
+    private final Map<Class<?>, Function<Object, M>> constructorCache = new ConcurrentHashMap<>();
+
+    public ConstructorBasedEventSourcedEntityCreator(Class<M> entityType) {
```

Comment:
I am missing JavaDoc on the constructor, as well as the `@Nonnull` annotation and it's validation on the parameter.

Reply:
Javadoc added